### PR TITLE
Add Metric Collection similar to moto tests

### PR DESCRIPTION
**Pull request by baermat**
This PR adds metric collection to the terraform tests like we have them for the moto tests we run against LocalStack. This includes the following changes:
- The default image to test against is now `localstack/localstack-pro`
- All applicable services are tested, not just community services anymore
- LocalStack is now started via `localstack` cli command in the pipeline
- Services that take too long are divided into partitions to avoid timeouts. Right now this includes ecs, ec2, and elasticache. This service-partition list is not created automatically as of this point. All other service have one partition named 'All'
- Removed the test parallelization for ec2. Doubled the number of initial partitions in response
- All metrics are collected in csv files and uploaded in the final artifacts. One file per partition is created
- Replace the option to specify arbitrary images with a switch whether to enable the pro image or not (in which case community is used)

### CODE REVIEW

File: terraform_pytest/service-partitions.yml [hunk 1]
```diff
@@ -0,0 +1,1389 @@
+# This file partitions services that take too long in one go to
```
Comment (macnev2013) [inline terraform_pytest/service-partitions.yml:1]: How are we generating this file? Do we have any utility for this?

**Review by macnev2013 — CHANGES_REQUESTED** [review]
Great set of changes 🚀 Only have one minor doubt else LGTM 👍🏻

File: terraform_pytest/service-partitions.yml (hunk 1 above)
Comment (baermat) [reply terraform_pytest/service-partitions.yml:1]: No, right now it is created manually by gathering the tests of the affected services via `--collect-only` and then partitioning it by hand based on the test durations and results. Creating and partitioning this automatically is something for a future iteration. @alexrashed mentioned that he is working on something with `pytest split` that could alleviate this problem, but this is still a work in progress. I will put a more detailed comment on how this file is created in, if that works for you.

Comment (macnev2013) [reply terraform_pytest/service-partitions.yml:1]: Thanks for the explanation 👍🏻

**Review by macnev2013 — APPROVED** [review]

File: terraform_pytest/service-partitions.yml (hunk 1 above)
Comment (alexrashed) [reply terraform_pytest/service-partitions.yml:1]: Just chiming in here to clarify the last statement:
It depends a bit on how the partitions are created. We currently don't really have a nice and clean way of creating time-balanced partitions automatically, but [pytest-split](https://jerry-git.github.io/pytest-split/) (which is an open-source pytest plugin which we already use in some GitHub action pipelines) simplifies the creation of naive split partitions, i.e. creating equally-sized chunks. Or there would also be the possibility to commit duration data into the repository, but these wouldn't be updated automatically.

Nevertheless, I think changes towards `pytest-split` could easily be done in a later iteration.

File: README.md [hunk 2]
```diff
@@ -33,11 +34,11 @@ Purpose of this project is to externalize the test cases from the localstack rep
...
-- `--ls-start`: Start localstack instance before running the test cases
-- `--ls-image`: Specify the localstack image to use, default is `localstack/localstack:latest`
\ No newline at end of file
+- `--ls-start`: Start localstack instance before running the test cases. Will use the cli by running `localstack start -d`
+- `--gather-metrics`: Collects raw test metrics for the run. Requires manual installation of the extension first:
+   ```bash
+    localstack extensions init
+    localstack extensions install "git+https://github.com/localstack/localstack-moto-test-coverage/#egg=collect-raw-metric-data-extension&subdirectory=collect-raw-metric-data-extension"
+   ```
+   Expects a `SERVICE` environment variable to be set for naming the metric file.
```
Comment (steffyP) [inline README.md:44]: NIT: maybe we could briefly mention that we have some services black-listed, and some on "ignore".

**Review by steffyP — APPROVED** [review]
This is amazing! 🥳 🎉 Great work also adding partitions for the services that took too long! 👍 
We now have so much more data than before, and I am really excited to include the results in the docs 😄 
Guess a lot of customers will be happy as well to get this additional insights 🚀

File: README.md (hunk 2 above)
Comment (baermat) [reply README.md]: :+1: added some info in the README

**Pull request merged**